app/grandchallenge/workstations/static/workstations/js/session-control.js

```diff
@@ -6,14 +6,28 @@ function openWorkstationSession(element) {
         const url = element.dataset.createSessionUrl;
         const query = element.dataset.workstationQuery;
         const creationURI = `${url}?${query}`;
+        const sessionListUrl = element.dataset.sessionUrl;
+        const domain = element.dataset.domain;
 
         if (event.ctrlKey) {
             window.open(creationURI);
             return;
         }
 
-        // PoC Hack: ToDo query any existing session
-        const sessionOrigin = element.dataset.sessionOrigin;
+        let activeSessions;
+        $.ajax({
```

Comment:
This adds an API call on each click, which does a filter on the values in the db. Given we only have at most 3 active regions, isn't it quicker just to broadcast the message client side to each of the active regions in turn?